### PR TITLE
Closes #19090: Show snackbar on adding to bookmarks from tabs tray

### LINKED ISSUE #19090: Show snackbar on adding to collection/bookmark from tabs tray
This is a regression in the behaviour from the old tabs tray.

We need to add the two different collection snackbars:
 1. Adding to an existing collection.
 2. Adding to a new one.

UI tests to re-enable when fixed:
 - `verifyExpandedCollectionItemsTest`
 - `shareCollectionTest`
 - `deleteCollectionTest`

See the old tabs tray for reference for now:

https://user-images.githubusercontent.com/1370580/115662746-6cc9d100-a30d-11eb-886d-57896d90fa03.mp4

### PR DESCRIPTION
**Pull request by rocketsroger**
### Pull Request checklist

- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Comment by jonalmeida** [comment]
Can we re-enable the UI tests that were previously failing before this change? `createFirstCollectionTest`, `shareCollectionTest`, `deleteCollectionTest`

File: app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt [hunk 1]
```diff
@@ -430,6 +433,28 @@ class TabsTrayFragment : AppCompatDialogFragment() {
...
 
+    @VisibleForTesting
+    internal fun showBookmarkSnackbar(
+        tabSize: Int
+    ) {
+        val anchor = if (requireComponents.settings.accessibilityServicesEnabled) {
+            null
+        } else {
+            new_tab_button
+        }
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt:444]: Could we put this is a common getter for `showCollectionSnackbar` to also share?

**Comment by rocketsroger** [comment]
> Can we re-enable the UI tests that were previously failing before this change? `createFirstCollectionTest`, `shareCollectionTest`, `deleteCollectionTest`

Ah, forgot to do this.  I'll check.  Thanks,

**Review by jonalmeida — COMMENTED** [review]
Looks good, let's enable those UI tests in this change so that we can verify there are no regressions in expected behaviour. 👌

**Comment by rocketsroger** [comment]
> > Can we re-enable the UI tests that were previously failing before this change? `createFirstCollectionTest`, `shareCollectionTest`, `deleteCollectionTest`
> 
> Ah, forgot to do this. I'll check. Thanks,

This looks unrelated to the issue reported though.  The issue is about the bookmark snackbar.  These UI tests are collection related that was added recently.

File: app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt (hunk 1 above)
Comment (rocketsroger) [reply app/src/main/java/org/mozilla/fenix/tabstray/TabsTrayFragment.kt:444]: Good idea.  I'll refactor.

**Comment by rocketsroger** [comment]
ui tes

File: app/src/main/java/org/mozilla/fenix/tabstray/NavigationInteractor.kt [hunk 2]
```diff
@@ -171,12 +172,12 @@ class DefaultNavigationInteractor(
 
     override fun onSaveToCollections(tabs: Collection<Tab>) {
         metrics.track(Event.TabsTraySaveToCollectionPressed)
+        tabsTrayStore.dispatch(TabsTrayAction.ExitSelectMode)
```
Comment (rocketsroger) [inline app/src/main/java/org/mozilla/fenix/tabstray/NavigationInteractor.kt:175]: Found that this is causing the race between showing the floating new tab button and showing the snackbar.  Moving this so we exit the select mode when the dialog is shown removes the possibility of that race.

This doesn't change the behaviour for both positive or negative case.

**Review by jonalmeida — APPROVED** [review]

File: app/src/main/java/org/mozilla/fenix/tabstray/NavigationInteractor.kt (hunk 2 above)
Comment (jonalmeida) [reply app/src/main/java/org/mozilla/fenix/tabstray/NavigationInteractor.kt:175]: Thank you for figuring this out! Sorry it turned out to be unrelated.

**Pull request merged**